src/main/scala/akka/contrib/process/NonBlockingProcess.scala

```diff
@@ -223,34 +169,27 @@ class NonBlockingProcess(
             .foreach[ByteString](bytes => nuProcess.writeStdin(bytes.toByteBuffer))
         val (out, stdout) =
           Source
-            .fromGraph(new PublishIfAvailable[ByteString])
+            .queue[ByteString](stdoutBufferMaxLines, OverflowStrategy.dropHead)
```

Comment:
...dropping the oldest element is inline with what we've been doing with our Elasticsearch logging. Seems like a reasonable policy here.